### DISCUSSION TITLE
v0.14.0-rc.0

### DISCUSSION THREAD

**Discussion by rfratto** [grafana-cold-storage/agent#520] [General]
This is our first release candidate release of the Grafana Agent! A lot has changed over the past month. In fact, this release is so big that we wanted to do a release candidate first to gain more confidence in the code. Please try this release out in non-production environments and let us know if you run into any problems!

We're also going to start marking features interchangeably as *beta* or *experimental* when we believe they are going to change. This is to help us continue iterating quickly while making it clear which things we believe are production-ready. 

### Upgrading
⚠️ **BREAKING CHANGES** ⚠️

This release updates to Prometheus 2.26.0, which includes the SigV4 code we initially introduced here 🎉 However, the upstream SigV4 code introduced a **breaking change** from the initial Grafana Agent SigV4 configuration. Please read the [migration guide](https://github.com/grafana/agent/blob/v0.14.0-rc.0/docs/migration-guide.md) for specific instructions on how to change your SigV4 config to work with the new release. 

### Notable changes:
This is the first release where we have changed the name of the project from **Grafana Cloud Agent** to **Grafana Agent**. The new name is for a few reasons, but I (@rfratto) like to say "Grafana Agent" is less confusing, since it was never intended to only be used by Grafana Cloud users. The README has been updated to reflect who might find the Grafana Agent useful. 

With that out of the way, let's get into the biggest changes of this release: 

* Thanks to the work by @mattdurham, official Windows support is now in beta! The Grafana Agent now supports being run natively as a Windows service, logs to the Windows Event Log, has a Windows isntaller, and includes windows_exporter as an integration. 
* Also in beta is `/-/reload`, where you can tell an Agent to dynamically reload its entire config file. Paired with this is a `/-/config` endpoint to return the current in-memory config being used. 
* More things in beta! Courtesy of @yeya24, you can now generate metrics from your spans!
* Thanks to @mattdurham, we finally support a global remote_write block! No more duplicating remote_write across both integrations and scrape configs!  
* The FreeBSD build returns, as well as official support for M1 Macs!

There's a ton of changes this time around, so check out the full [CHANGELOG](https://github.com/grafana/agent/blob/v0.14.0-rc.0/CHANGELOG.md#main-unreleased) for all the details.

### Installation:
Grafana Agent is currently distributed in plain binary form, Docker
container images, and a Kubernetes install script. Choose whichever fits your
use-case best.

#### Kubernetes Install Script

The following scripts will download and install two Kubernetes manifests for the
Agent. The first manifest collects metrics, the second collects logs, and the
final collects traces. You will be prompted for input for each manifest. The
script requires curl and envsubst (GNU gettext).

```
NAMESPACE="default" /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/grafana/agent/v0.14.0-rc.0/production/kubernetes/install.sh)" | kubectl apply -f -
NAMESPACE="default" /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/grafana/agent/v0.14.0-rc.0/production/kubernetes/install-loki.sh)" | kubectl apply -f -
NAMESPACE="default" /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/grafana/agent/v0.14.0-rc.0/production/kubernetes/install-tempo.sh)" | kubectl apply -f -
```

#### Docker container:

* https://hub.docker.com/r/grafana/agent

```bash
docker pull "grafana/agent:v0.14.0-rc.0"
```

#### Binary

We provide precompiled binary executables for the most common operating systems.
Choose from the assets below for your matching operating system. Example for the
`linux` operating system on `amd64`:

```bash
# download the binary
curl -O -L "https://github.com/grafana/agent/releases/download/v0.14.0-rc.0/agent-linux-amd64.zip"

# extract the binary
unzip "agent-linux-amd64.zip"

# make sure it is executable
chmod a+x "agent-linux-amd64"
```

#### `agentctl`

`agentctl`, a tool for helping you interact with the Agent,
is available as a Docker image:

```bash
docker pull "grafana/agentctl:v0.14.0-rc.0"
```

Or as a binary. Like before, choose the assets below that matches your
operating system. For example, with `linux` on `amd64`:

```bash
# download the binary
curl -O -L "https://github.com/grafana/agent/releases/download/v0.14.0-rc.0/agentctl-linux-amd64.zip"

# extract the binary
unzip "agentctl-linux-amd64.zip"

# make sure it is executable
chmod a+x "agentctl-linux-amd64"
```

**Comment by krozlink** [comment]
where can I find the new windows installer mentioned in this release? I can't see it in the list of assets

**Reply by rfratto** [reply]
Ah, thanks, this looks like a bug and we'll need to fix it in the next rc. I opened up #521 to track.

**Reply by rfratto** [reply]
v0.14.0-rc.2 has just been released and includes the Windows installer. Thanks for letting us know about this!
